aea/test_tools/test_cases.py

```diff
@@ -297,35 +297,64 @@ def difference_to_fetched_agent(cls, public_id: str, agent_name: str) -> List[st
         # for pydocstyle
         def is_allowed_diff_in_agent_config(
             path_to_fetched_aea: str, path_to_manually_created_aea: str
-        ) -> Tuple[bool, Dict[str, str], Dict[str, str]]:
+        ) -> Tuple[
+            bool, Union[Dict[str, str], List[Any]], Union[Dict[str, str], List[Any]]
+        ]:
             with open_file(
                 os.path.join(path_to_fetched_aea, "aea-config.yaml"), "r"
             ) as file:
-                content1 = list(yaml.safe_load_all(file))[0]  # only load first page
+                content1 = list(yaml.safe_load_all(file))  # load all contents
             with open_file(
                 os.path.join(path_to_manually_created_aea, "aea-config.yaml"), "r"
             ) as file:
-                content2 = list(yaml.safe_load_all(file))[0]
-            content1c = copy.deepcopy(content1)
-            for key, value in content1c.items():
-                if content2[key] == value:
-                    content1.pop(key)
-                    content2.pop(key)
+                content2 = list(yaml.safe_load_all(file))
+
+            content1_agentconfig = content1[0]
+            content2_agentconfig = content2[0]
+            content1_agentconfig_copy = copy.deepcopy(content1_agentconfig)
+
+            # check only agent part
+            for key, value in content1_agentconfig_copy.items():
+                if content2_agentconfig[key] == value:
+                    content1_agentconfig.pop(key)
+                    content2_agentconfig.pop(key)
             allowed_diff_keys = [
                 "aea_version",
                 "author",
                 "description",
                 "version",
+                "connection_private_key_paths",
```

Comment:
Why do they need to be added?

Reply:
I see, just for convenience

Reply:
yes. Also, don't you think it makes more sense? O/w we could add keys after the comparison with fetched agent